TheraLang.BLL/Services/UserManagementService.cs

```diff
@@ -27,15 +27,14 @@ public class UserManagementService : IUserManagementService
         }
 
         public async Task<User> GetUser(string email, string password)
-        {
-            var user = await _unitOfWork.Repository<User>().GetAll().Include(x => x.Role)
-                .FirstOrDefaultAsync(u => u.Email == email);
-            if (PasswordHasher.VerifyHashedPassword(user.PasswordHash, password))
-            {
-                return user;
-            }
-
-            throw new Exception($"Cannot get user with {nameof(email)}: {email}.");
+        { 
+                var user = await _unitOfWork.Repository<User>().GetAll().Include(x => x.Role)
+                    .FirstOrDefaultAsync(u => u.Email == email);
+                if (PasswordHasher.VerifyHashedPassword(user.PasswordHash, password))
+                {
+                    return user;
+                }
+                throw new NotFoundException($"Cannot get user with {nameof(email)}: {email}.");
```

Comment:
Check NotFoundException constuctor, your probably should change your message to fit it.